cloud/db/organizations.ts

```diff
@@ -558,6 +561,34 @@ export class Organizations extends BaseAuthenticatedEffectService<
         organizationId,
       });
 
+      // Fetch organization to get stripeCustomerId
+      const [org] = yield* client
+        .select(publicFields)
+        .from(organizations)
+        .where(eq(organizations.id, organizationId))
+        .limit(1)
+        .pipe(
+          Effect.mapError(
+            (e) =>
+              new DatabaseError({
+                message: "Failed to fetch organization for deletion",
+                cause: e,
+              }),
+          ),
+        );
+
+      if (!org) {
+        return yield* Effect.fail(
+          new NotFoundError({
+            message: `organization with organizationId ${organizationId} not found`,
+            resource: this.getResourceName(),
+          }),
+        );
+      }
+
+      // Cancel all active subscriptions before deleting
+      yield* cancelSubscriptions(org.stripeCustomerId);
```

Comment:
Needs tests checking this is invoked with the correct customerId. Probably worth covering the StripeError case too.